torch/distributed/_sharded_tensor/api.py

```diff
@@ -35,6 +35,25 @@ class Shard(object):
     tensor: torch.Tensor
     metadata: ShardMetadata
 
+@dataclass
+class ShardedTensorMetadata(object):
+    """
+    Represents metadata for :class:`ShardedTensor`
+    """
+
+    # Metadata about each shard of the Tensor
+    shards_metadata: List[ShardMetadata]
+
+    # Size of each dim of the overall Tensor.
+    overall_dims: torch.Size
```

Comment:
since this is already a sharded tensor metadata, should we just call this as `size` to match with other fields?